packages/cap-config-writer/src/cap-writer/package-json.ts

```diff
@@ -0,0 +1,130 @@
+ * @param {string} appId - The ID of the app.
+ * @param {boolean} [useNPMWorkspaces] - Whether to use npm workspaces.
+ * @returns {{ [x: string]: string }} The CDS watch script for the CAP app.
+ */
+export function getCDSWatchScript(
+    projectName: string,
+    appId: string,
+    useNPMWorkspaces: boolean = false
+): { [x: string]: string } {
+    const DisableCacheParam = 'sap-ui-xx-viewCache=false';
+    // projects by default are served base on the folder name in the app/ folder
+    // If the project uses npm workspaces (and specifically cds-plugin-ui5 ) then the project is served using the appId including namespace
+    const project = useNPMWorkspaces ? appId : projectName + '/webapp';
+    return {
+        [`watch-${projectName}`]: `cds watch --open ${project}/index.html?${DisableCacheParam}${
+            useNPMWorkspaces ? ' --livereload false' : ''
+        }`
+    };
+}
+
+/**
+ * Updates the scripts in the package json file with the provided scripts object.
+ *
+ * @param {Editor} fs - The file system editor.
+ * @param {string} packageJsonPath - The path to the package.json file.
+ * @param {Record<string, string>} scripts - The scripts to be added or updated in the package.json file.
+ * @returns {void}
+ */
+function updatePackageJsonWithScripts(fs: Editor, packageJsonPath: string, scripts: Record<string, string>): void {
+    fs.extendJSON(packageJsonPath, { scripts });
+}
+
+/**
+ * Updates the scripts in the package json file for a CAP project.
+ *
+ * @param {Editor} fs - The file system editor.
+ * @param {string} packageJsonPath - The path to the package.json file.
+ * @param {string} projectName - The name of the project.
+ * @param {string} appId - The ID of the app.
+ * @param {boolean} [enableNPMWorkspaces] - Whether to enable npm workspaces.
+ * @param {Logger} [log] - The logger instance for logging warnings.
+ * @returns {Promise<void>} A Promise that resolves once the scripts are updated.
+ */
+async function updateScripts(
+    fs: Editor,
+    packageJsonPath: string,
+    projectName: string,
+    appId: string,
+    enableNPMWorkspaces?: boolean,
+    cdsUi5PluginInfo?: CdsUi5PluginInfo
+    //log?: Logger
+): Promise<void> {
+    const packageJson = (fs.readJSON(packageJsonPath) ?? {}) as Package;
+    const hasNPMworkspaces = await checkCdsUi5PluginEnabled(packageJsonPath, fs);
+    // Determine whether to add cds watch scripts for the app based on the availability of minimum CDS version information.
+    // If 'cdsUi5PluginInfo' contains version information and it satisfies the minimum required CDS version,
+    // or if 'cdsUi5PluginInfo' is not available and the version specified in 'package.json' satisfies the minimum required version,
+    // then set 'addScripts' to true. Otherwise, set it to false.
+    const addScripts = cdsUi5PluginInfo?.hasMinCdsVersion ? cdsUi5PluginInfo.hasMinCdsVersion : satisfiesMinCdsVersion(packageJson)
+    if(addScripts) {
```

Comment:
Here we add cds watch scripts only in cdsUi5PluginInfo has minimum cds version available, this is set during the prompt phase of the app generation. Question for head less app generation - according to the logic here I simply read from the package.json, is this consistent ??? Is there a better way to cater for this watch script addition for headless app?